crates/astria-grpc-mock-test/tests/health/test_response.rs

```diff
@@ -0,0 +1,137 @@
+        .respond_with(response::default_response::<HealthCheckResponse>());
+    server.mocked.register(mock).await;
+    let rsp = client
+        .check(HealthCheckRequest {
+            name: "helloworld".to_string(),
+            service: "helloworld".to_string(),
+        })
+        .await
+        .unwrap();
+    assert_eq!(&HealthCheckResponse::default(), rsp.get_ref());
+}
+
+#[tokio::test]
+async fn constant_response_works() {
+    let server = start_mock_server().await;
+    let mut client = HealthClient::connect(format!("http://{}", server.local_addr))
+        .await
+        .unwrap();
+    let expected_response = HealthCheckResponse {
+        status: 1,
+    };
+    let mock = Mock::for_rpc_given("check", matcher::message_type::<HealthCheckRequest>())
+        .respond_with(response::constant_response(expected_response.clone()));
+    server.mocked.register(mock).await;
+    let rsp = client
+        .check(HealthCheckRequest {
+            name: "helloworld".to_string(),
+            service: "helloworld".to_string(),
+        })
+        .await
+        .unwrap();
+    assert_eq!(&expected_response, rsp.get_ref());
+}
+
+#[tokio::test]
+async fn dynamic_response_works() {
+    let server = start_mock_server().await;
+    let mut client = HealthClient::connect(format!("http://{}", server.local_addr))
+        .await
+        .unwrap();
+    let mut expected_response = HealthCheckResponse {
+        status: 1,
+    };
+
+    let mock = Mock::for_rpc_given("check", matcher::message_type::<HealthCheckRequest>())
+        .respond_with(response::dynamic_response(dynamic_responder));
+    server.mocked.register(mock).await;
+    let rsp_1 = client
+        .check(HealthCheckRequest {
+            name: "helloworld".to_string(),
+            service: "1".to_string(),
+        })
+        .await
+        .unwrap();
+    assert_eq!(&expected_response, rsp_1.get_ref());
+
+    expected_response.status = 2;
+    let rsp_2 = client
+        .check(HealthCheckRequest {
+            name: "helloworld".to_string(),
+            service: "2".to_string(),
+        })
+        .await
+        .unwrap();
+    assert_eq!(&expected_response, rsp_2.get_ref());
+}
+
+fn dynamic_responder(request: &HealthCheckRequest) -> HealthCheckResponse {
+    let status_return = request
+        .service
+        .chars()
+        .next()
+        .unwrap()
+        .to_digit(10)
+        .unwrap();
+    HealthCheckResponse {
+        status: i32::try_from(status_return).unwrap(),
+    }
+}
+
+#[tokio::test]
+async fn response_delay_works_as_expected() {
+    let server = start_mock_server().await;
+    let mut client = HealthClient::connect(format!("http://{}", server.local_addr))
+        .await
+        .unwrap();
+    let mock = Mock::for_rpc_given("check", matcher::message_type::<HealthCheckRequest>())
+        .respond_with(
+            response::default_response::<HealthCheckResponse>()
+                .set_delay(Duration::from_millis(250)),
```

Comment:
I'd maybe make this duration a `const` to be used here and below.

Reply:
After implementing Janis' suggestion below, I'm not sure this is necessary anymore, but happy to put it in if you think it is!

Reply:
I think it's still worthwhile, since we can specify e.g.
```rust
const DELAY: Duration = Duration::from_millis(250);
const FIFTY_MILLIS: Duration = Duration::from_millis(50);
```
then below use, `DELAY` to set the delay and `DELAY - FIFTY_MILLIS` for the error case and `DELAY + FIFTY_MILLIS` for the success one.

I think that's slightly easier to understand and update if needed.

Reply:
Done in 7cebad8